examples/summarization/scripts/reward_summarization.py

```diff
@@ -0,0 +1,182 @@
+            self.config.hidden_size
+            if hasattr(self.config, "hidden_size")
+            else self.config.n_embd
+        )
+        self.transformer = model.transformer
+        self.v_head = nn.Linear(self.config.n_embd, 1)
+
+    def forward(
+        self,
+        input_ids_j=None,
+        attention_mask_j=None,
+        input_ids_k=None,
+        attention_mask_k=None,
+        return_loss=True,
+    ):
+        transformer_outputs_j = self.transformer(
+            input_ids_j,
+            attention_mask=attention_mask_j,
+        )
+        hidden_states_j = transformer_outputs_j[0][:,-1,:]  # Get the last hidden state
+        rewards_j = self.v_head(hidden_states_j)
+        if input_ids_k is not None:
+            transformer_outputs_k = self.transformer(
+                input_ids_k,
+                attention_mask=attention_mask_k,
+            )
+            hidden_states_k = transformer_outputs_k[0][:,-1,:]  # Get the last hidden state
+            rewards_k = self.v_head(hidden_states_k)
+
+            loss = -nn.functional.logsigmoid(rewards_j - rewards_k).mean()
+            return {"loss": loss, "rewards_j": rewards_j, "rewards_k": rewards_k}
+        return {"rewards_j": rewards_j}
+
+# Define the training args. Needs to be done before the model is loaded if you are using deepspeed.
+training_args = TrainingArguments(
+    output_dir=f"{script_args.gpt_model_name}_summarization_reward_model",
+    learning_rate=script_args.learning_rate,
+    per_device_train_batch_size=script_args.per_device_train_batch_size,
+    per_device_eval_batch_size=script_args.per_device_eval_batch_size,
+    num_train_epochs=5,
+    weight_decay=0.001,
+    evaluation_strategy="epoch",
+    save_strategy="epoch",
+    gradient_accumulation_steps=script_args.gradient_accumulation_steps,
+    load_best_model_at_end=True,
+    push_to_hub=True,
+    deepspeed=script_args.deepspeed,
+    local_rank=script_args.local_rank,
+)
+
+# Load the model and tokenizer.
+tokenizer = AutoTokenizer.from_pretrained(script_args.gpt_model_name)
+tokenizer.pad_token = tokenizer.eos_token
+model = GPTRewardModel(script_args.gpt_model_name)
+
+# Turn the dataset into our special j (preferred summary) and k (not preferred summary) format.
+def turn_into_text_classification_format(examples):
+    
+    new_examples = {"text_j": [], "text_k": []}
+
+    for info, summaries, choice in zip(examples["info"], examples["summaries"], examples["choice"]):
+        assert len(summaries) == 2
+        assert choice in (0,1)
+        original_text_field = "post" if info["post"] is not None else "article"
+        new_examples["text_j"].append(summaries[choice]["text"] + " " + tokenizer.bos_token + " " + info[original_text_field])
+        new_examples["text_k"].append(summaries[0 if choice==1 else 1]["text"] + " " + tokenizer.bos_token + " " + info[original_text_field])
+
+    return new_examples
+
+num_proc = 100 # Can adjust to be higher if you have more processors.
```

Comment:
Is this for different distributed training architectures?

Reply:
no it's just an argument that you can add in the datasets map function https://huggingface.co/docs/datasets/process

It should work even if your system doesn't have 100 cpus

Reply:
Cool, maybe add a comment saying that (good plug to HF tools)

Reply:
done!